analysis_scripts/qc_scripts/bg_springer_qc.Rmd

```diff
@@ -0,0 +1,210 @@
+---
+title: "bg_springer_qc"
+author: "Natalie Davidson"
+date: "3/29/2021"
+output: rmarkdown::github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+
+require(data.table)
+require(here)
+require(ggplot2)
+require(stringr)
+
+
+proj_dir = here()
+source(file.path(proj_dir, "/utils/scraper_processing_utils.R"))
+source(file.path(proj_dir, "/process_doi_data/springer_scripts/springer_scrape_utils.R"))
+
+```
+
+
+## Overview
+
+This notebook will QC the background Springer articles.
+The method of getting the Springer articles are to: 
+**1)** get author names from a random selection of articles.
+200 articles per month, from 2005-2020.
+**2)** From these random articles, take a subset in order to get the author affiliation information.
+The query for author affiliation is only a per country sum over the articles, so unlike the author names were we can get the information of 50 articles per API call, to get country affiliations we can only do 1 API call per article. 
+This forces us to have less articles with country affiliations. 
+**3)** From the author names in step 1 we predict gender.
+**4)** From the countries in step 2, we standardize the country names.
+
+The specific files for each step of the pipeline:
+
+1. author information from the randomly chosen Springer articles: `/data/reference_data/springer_bg_author_cache.tsv`
+
+2. author countries from a random subset of articles from step 1: `/data/reference_data/springer_random_country_cache.tsv`
+
+3. gender predictions: `/data/author_data/springer_author_gender.tsv`
+
+4. location predictions: `data/author_data/all_author_country.tsv`
+
+
+## Pipeline Step 1: Author Names from API
+
+
+```{r step1_read, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# read in the API responses from springer
+pipeline_1_file = file.path(proj_dir,
+                    "/data/reference_data/springer_bg_author_cache.tsv")
+author_name_df = fread(pipeline_1_file)
+
+
+# plot number of articles selected
+ggplot(unique(author_name_df[,c("doi", "year", "month")]), aes(x=as.factor(year))) +
+    geom_bar() + theme_bw() + geom_hline(yintercept = 2400, color="red") +
+    xlab("Year of Article") + ylab("# articles") +
+        ggtitle("# Articles queried with Author Names Over Time")
```

Comment:
The number of articles is constant over all years?

Reply:
In this case, yes, because I am taking a random sample for a fixed size.